src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -1398,10 +1398,31 @@ private void visitVarDeclaration(String name, JSType type) {
     }
 
     private void visitTemplateTypes(ObjectType type) {
-      visitTemplateTypes(type, Collections.<String>emptySet());
+      visitTemplateTypes(type, Collections.<String>emptySet(), true);
     }
 
-    private void visitTemplateTypes(ObjectType type, Set<String> alreadyEmittedTemplateType) {
+    /**
+     * Emits template types for a given type. For example for
+     *
+     * <pre>T<A,B></pre>
+     *
+     * , this method will emit
+     *
+     * <pre><A,B></pre>
+     *
+     * .
+     *
+     * @param type the type in question
+     * @param alreadyEmittedTemplateType in certain scopes some template types will be already
```

Comment:
maybe you could mention that this is for `@template` on methods that repeat an `@template` of the class (is it?)

Reply:
done